### PR TITLE
New framework for setting and using configuration data

### PR DESCRIPTION
**Pull request by RayPlante**
This PR is part of a larger effort to fix server-side rendering.  Currently, exceptions occurring on the server-side are causing rendering to stop soon after encoding the NERDm JSON data into the landing page (via Angular's TransferState).  This larger effort is aimed at eliminating those errors as well as at making it simpler to make robust improvements going forward.  

In this step, this PR establishes a new framework for handling and using configuration information.  It takes into account the different runtime contexts (local development vs testing vs. production operation) and is described in the README.md file in [angular/src/app/config](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/app/config).  In particular, during development and testing, one will want to set the configuration via a local file; in production, the data must come from the configuration service, be used by the server but also transmitted to the browser to configure the browser-side application.

The new framework is implemented via a new (sub-) module in [app/config](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/app/config) (replacing the `config-service` in [app/shared](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/app/shared)).  When running the application in development (either as a browser-only app or with server-side rendering), the configuration is set via [src/enivronments/envirtonment.ts](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/environments/environment.ts).  When running under oar-docker, the server-side application reads the configuration from a file whose location is set by an environment variable.  (This is file is written and the environment variable set by the docker container that hosts the landing page service.)  The service side then transmits the configuration along with the rendered page using the Angular TransferState mechanism.  

The new framework introduces a new schema for the landing page service.  It will be retrieved from the config server via the name `pdr-lps`.  It adds some additional configuration parameters and renames others.  Documentation for the schema can be found in-line in [app/config/config.ts](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/app/config/config.ts).

### CODE REVIEW

File: angular/src/app/landing/landing.component.html [hunk 1]
```diff
@@ -144,20 +145,23 @@ <h2>{{record.title}}</h2>
...
               </span>
             </div>
           </div>
           <div class="ui-g">
             <div class="ui-g-12 ui-md-12 ui-lg-12 ui-sm-12">
               <description-resources [record]="record" [files]="files" [distdownload]="distdownload"
-                [isInternal]="isInternal" [filescount]="filescount" [metadata]="metadata"></description-resources>
+                                     [isInternal]="isInternal" [filescount]="filescount" [inBrowser]="inBrowser"
+                                     [metadata]="metadata"></description-resources>
               <metadata-detail [record]="record" [serviceApi]="serviceApi" [metadata]="metadata"></metadata-detail>
```
Comment (chuanlin2018) [inline angular/src/app/landing/landing.component.html]: Should we add [inBrowser]="inBrowser" here as well?

Comment (chuanlin2018) [reply angular/src/app/landing/landing.component.html]: I understand that. I meant line 157 - I saw the MetadataComponent had @Input() inBrowser but seems the param was missing in line 157...

**Comment by chuanlin2018** [comment]
DId you mean added "editEnabled" field to the record in the database?

File: angular/src/app/landing/landing.component.html (hunk 1 above)
Comment (RayPlante) [reply angular/src/app/landing/landing.component.html]: Ah, I see--sorry about that!  (The display did not make it obvious which line you were talking about.)  

Yes, you are right!!  It should be there; I'll add it real quick.  Thanks!

Comment (chuanlin2018) [reply angular/src/app/landing/landing.component.html]: I enjoy reading your code :) very helpful for understanding server side rendering.

**Comment by RayPlante** [comment]
No, sorry--I meant as a configuration parameter (as defined in [app/config/config.ts](/usnistgov/oar-pdr/tree/feature/newconfig/angular/src/app/config/config.ts) and made accessible to the app via an `AppConfig` instance).

File: angular/src/app/landing/landing.component.html (hunk 1 above)
Comment (RayPlante) [reply angular/src/app/landing/landing.component.html]: Thanks for the careful review!

**Comment by chuanlin2018** [comment]
Just noticed "editEnabled" in the code. I will think about it.
For the endpoint for the editing service, yes I think we need that.

**Comment by RayPlante** [comment]
FYI, oardev was updated with latest version.

**Pull request merged**